test/libsolidity/semanticTests/events/event_anonymous_with_signature_collision.sol

```diff
@@ -0,0 +1,11 @@
+contract ClientReceipt {
+    event Deposit(uint256 indexed _from, bytes32 indexed _id, uint _value) anonymous;
+    function deposit(bytes32 _id) public payable {
+        emit Deposit(0x2012159ca6b6372f102c535a4814d13a00bfc5568ddfd72151364061b00355d1, _id, msg.value); // 0x2012159c -> 'Deposit(uint256,bytes32,uint256)'
+    }
+}
```

Comment:
I was thinking about something more along these lines:
```suggestion
contract ClientReceipt {
    event Withdrawal(uint _value, string owner);
    event Deposit(uint256 indexed _from, bytes32 indexed _id, uint _value) anonymous;
    function deposit(bytes32 _id) public payable {
        emit Deposit(0xd386301af7e1425796c2fb2be8fb727c237c76f20144e6488c5715a08954a5df, _id, msg.value); // 0x2012159c -> 'Deposit(uint256,bytes32,uint256)'
    }
}
```

Not sure if I got the hash right. I ran `echo -n "Withdrawal(uint256,string)" | keccak-256sum`. The point is to get the hash of the signature of a non-anonymous event into an argument of an anonymous one.

Reply:
Ah, ok now I understood what you mean. I think we cannot do anything about such collisions. It is always possible to create an anonymous event to fake the signature of another event. Because it is not really possible to fully detect this situation, I just removed that assertions completely:
```
if (eventSignature.has_value())
    soltestAssert(eventSignature.value().indexedTypes.size() == log.topics.size() - 1, "");
```
from `SemanticTest::eventSideEffectHook(..)` now.

Now also `SemanticTest::formatParameter(..)` is used to format the parameters, where it only accesses the indexed/nonindexed data types from the actual log, not from a wrongly matched event.

Reply:
The hash was correct. But I renamed the event to `Withdraw(uint256,string)`.